### PR TITLE
keep test window hidden after toggle

### LINKED ISSUE #272: Use collapsed state by default
#### Issue Description
It is super annoying that the test suite thing overlaps while you are actually writing code, but not testing anything. Every reload will expand the test suite again when you collapsed it by hand. Therefor i suggest to set the collapsed state to collapsed by default or at least add a checkbox that can be toggled to do so.
Store the state in localStorage for example, thats totally sufficient for a session.

### PR DESCRIPTION
**Pull request by tbushman**
#### Pre-Submission Checklist

- [x] Your pull request targets the `master` branch of freeCodeCamp/testable-projects-fcc
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] Your changes have been tested either locally or using a newly created CDN based on your fork's testable-projects-fcc/build/bundle.js file
- [x] Working changes have been added to the build by running `npm run build`

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.

- [x] Closes currently open issue: Closes #272 

#### Description

Using the localStorage method suggested by @Knochenmark , keep the test window collapsed after the user hides it.

### CODE REVIEW

File: src/utils/test-suite-skeleton.html [hunk 1]
```diff
@@ -1,11 +1,11 @@
-<div id="fcc_test_suite_indicator_wrapper"></div>
 <div id="fcc_foldout_toggler">
 	<span id="hamburger_top" class="fcc_hamburger transform_top"></span>
 	<span id="hamburger_middle" class="fcc_hamburger transform_middle"></span>
 	<span id="hamburger_bottom" class="fcc_hamburger transform_bottom"></span>
 </div>
 <input id="toggle" onclick="FCC_Global.hamburgerTransform()" type="checkbox"
 	title="CTRL + SHIFT + O">
+<div id="fcc_test_suite_indicator_wrapper"></div>
```
Comment (tbushman) [inline src/utils/test-suite-skeleton.html]: moved to enable `#toggle:checked` successor selector

File: src/stylesheets/style.css [hunk 2]
```diff
@@ -380,6 +380,9 @@ div {
   left: -320px;
   transition: .5s ease-in-out;
 }
+#toggle:checked ~ #fcc_test_suite_indicator_wrapper {
```
Comment (tbushman) [inline src/stylesheets/style.css]: Successor selector

File: webpack.config.dev.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ module.exports = {
             {
               test: /\.css$/,
               exclude: /node_modules/,
-              // chained loaders: 
+              // chained loaders:
```
Comment (tbushman) [inline webpack.config.dev.js]: I don't think these stray spaces warrant a separate PR

**Comment by tbushman** [comment]
Travis builds are currently stopped (https://www.traviscistatus.com/incidents/c1s4dlyxd4lf)

**Comment by tchaffee** [comment]
@tbushman This definitely sounds like a good change to me. But I hesitate in making UX behavior changes without first getting feedback from at least one person on the freeCodeCamp team. @QuincyLarson do you foresee any problems with keeping the test suite UI collapsed on refresh once the user intentionally collapses it?

**Comment by QuincyLarson** [comment]
@tchaffee As long as it's obvious as to how to un-collapse the menu, then yes - I think keeping it closed is superior UX, since the user has already expressed their intent to have it collapsed.

**Comment by tbushman** [comment]
Updated menu toggler to be 1px heavier and fCC green

**Comment by Knochenmark** [comment]
@tbushman Maybe you should give the hamburger menu a background color too? I think currently the color of the icon is black or? I imagine if someone changed the background color of his html container in codepen black on black will be hard to see :) Same will apply to the fcc green though. So maybe add some background-color +border?

**Comment by tbushman** [comment]
Do you want it in cornflower blue?

**Comment by Knochenmark** [comment]
@tbushman why you deleted my comment?

**Comment by tbushman** [comment]
I accidentally deleted your comment after deleting mine.

**Comment by tbushman** [comment]
Updated menu toggler with background

File: src/index.js [hunk 4]
```diff
@@ -76,31 +77,41 @@ $(document).ready(function() {
         // projectName variable is defined in our example projects so the
         // correct test suite is automatically loaded. This sets default text
         // for <option> text and project indicator in top right corner.
+
         if (typeof projectName !== 'undefined') {
           projectNameLocal = projectName;
         }
 
+        let toggleElement = document.getElementById('toggle');
```
Comment (tchaffee) [inline src/index.js]: I'm surprised this element has not been namespaced to `fcc_`. Not your problem, and I think it's a non-issue if we can put the UI in its own iframe. But if we cannot, this is something we should go back and fix. Maybe open an issue so we don't forget just in case?

File: src/stylesheets/style.css [hunk 5]
```diff
@@ -396,9 +399,19 @@ div {
 #fcc_foldout_toggler {
   position: absolute;
   top: 20px;
-  left: 20px;
+  left: 15px;
   z-index: 99998;
 }
+#fcc_foldout_toggler_background {
```
Comment (tchaffee) [inline src/stylesheets/style.css]: Are these style changes important to the functionality of keeping the UI closed? If not, it would have been better to put it in a separate PR, but I'm not going to make a fuss about it. Just a reminder to try to keep PRs small and focused on one change so we can more easily reverse each one without the reversing the other, if there are bugs.

File: src/index.js [hunk 6]
```diff
@@ -276,6 +287,11 @@ export function hamburgerTransform() {
     document.getElementById('hamburger_bottom').classList.remove(
       'transform_bottom'
     );
+    let placeholder = document.getElementById('placeholder');
+    let visibleProjectName = placeholder.innerHTML.split(' ').join('');
+    // Once the student has hidden the test window, this localStorage variable
+    // keeps it hidden until manually toggled.
+    localStorage.setItem('fCC_' + visibleProjectName + '_hide', true);
```
Comment (tchaffee) [inline src/index.js]: I don't see where this gets set to false. What am I missing?

Also see my comment above about making the building of this string DRY.

File: src/index.js [hunk 7]
```diff
@@ -76,31 +77,41 @@ $(document).ready(function() {
...
-            'fcc_test_suite_indicator_wrapper'
-          ).innerHTML =
+          placeholder.innerHTML = projectTitleCase;
+          indicatorWrapper.innerHTML =
             '<span id=fcc_test_suite_indicator>FCC Test Suite: ' +
             `${projectTitleCase}</span>`;
         }
+        let visibleProjectName = placeholder.innerHTML.split(' ').join('');
+        // If this is the first time loading this project, show test window
+        if (!localStorage.getItem('fCC_' + visibleProjectName + '_hide')) {
```
Comment (tchaffee) [inline src/index.js]: Searching on `'fCC_' + visibleProjectName + '_hide'` is a bit awkward. Could you assign this string to a variable and use the same variable name everywhere? I think this could even deserve a function that builds the name so that logic is in one place. DRY principle.

File: webpack.config.dev.js (hunk 3 above)
Comment (tchaffee) [reply webpack.config.dev.js]: Agreed

**Comment by tchaffee** [comment]
@tbushman What is the status of this? It looks like there are still some outstanding changes to be made or discussed?

**Comment by tbushman** [comment]
@tchaffee Back when the iframe PR (#279) seemed like a sure thing, I was holding off to rebase this. I'll get back to this soon. Thanks.

File: src/stylesheets/style.css (hunk 5 above)
Comment (tbushman) [reply src/stylesheets/style.css]: The changes were in reference to the comment above by @QuincyLarson re: making it obvious how to un-collapse the menu. This is a bolder toggler with a faint background.

File: src/index.js (hunk 6 above)
Comment (tbushman) [reply src/index.js]: It doesn't get set to false. Once hidden, the window stays hidden until toggled for each page load.

**Comment by QuincyLarson** [comment]
@tbushman Have you had a chance to make any progress on this? Anything we can do to help clear a path for you?

**Comment by tbushman** [comment]
@quincylarson Sorry. Need to rebase this. It will be ready in a few hrs. Brb

**Comment by QuincyLarson** [comment]
@tbushman Awesome! Thanks for letting us know. Good luck with the rebase.

@tchaffee you are in the best position to QA this - could you take a look at it at your convenience?

**Comment by tchaffee** [comment]
@tbushman I'm ready to QA this. Are all your changes in now?

**Comment by tbushman** [comment]
Yes.

**Comment by tbushman** [comment]
rebased after merging #296 as well

File: src/index.js [hunk 8]
```diff
@@ -55,6 +55,12 @@ import createHeatMapTests from './project-tests/heat-map-tests';
 export const assert = chai.assert;
 
 let projectNameLocal = false;
+let localStorageByProject = function() {
+  const placeholder = document.getElementById('placeholder');
+  const visibleProjectName = placeholder.innerHTML.split(' ').join('');
+  console.log(visibleProjectName);
```
Comment (tchaffee) [inline src/index.js]: Oops, you should remove this `console.log`. Sorry if I missed this the first time around.

**Comment by tbushman** [comment]
@tchaffee OK

**Comment by tchaffee** [comment]
NIce job. Merging.

**Comment by QuincyLarson** [comment]
@tbushman Thanks for implementing this nice UX improvement! 

@tchaffee Thanks for QA'ing!
